packages/web-components/src/components/va-file-input-multiple/README.md

```diff
@@ -0,0 +1,399 @@
+- **Behavior**: Automatically adds new empty file input when files are added; removes entries when files are deleted
+
+#### `@State() valueAdded: boolean`
+
+- **Purpose**: The state serves as a **one-time initialization flag** to track whether files added via the `value` prop have already been added to the `files` state or not.
+- **Initial Value**: `false`
+- **Behavior**: Set to `true` after files from the `value` prop are processed in `componentDidRender()`
+- **Use Case**: Ensures files provided via the `value` prop are only added to state once during component initialization
+
+### Events
+
+#### Primary Event: `vaMultipleChange`
+
+This event will be emitted when any change to a file occurs.
+
+```typescript
+{
+  action: 'FILE_ADDED' | 'FILE_UPDATED' | 'FILE_REMOVED' | 'PASSWORD_UPDATE',
+  file: File,           // The file that triggered the event. 
+  state: FileDetails[]  // An array of objects representing the current state of all files including the file that triggered the event.
+}
+```
+
+The `FileDetails` interface is defined as:
+
+```typescript
+interface FileDetails {
+  file: File,
+  changed: Boolean,
+  password?: string
+}
+```
+
+Each `vaMultipleChange` event includes a `state` array where:
+
+- **`changed: true`**: Indicates which file was affected by the current action
+- **`changed: false`**: All other files in the current state
+- **Deletion Exception**: No `changed` property is set during `FILE_REMOVED` actions
+
+```typescript
+// Example: A new file added
+state: [
+  { file: existingFile1, changed: false, password: undefined },
+  { file: newFile, changed: true, password: undefined },  // ← This slot received a file
+  { file: null, changed: false, password: undefined }     // ← New empty slot added ready for a file
+]
+```
+
+The File object of the file that triggered the change event is available in both the `file` property and the `state` array. This provides both granular (what changed) and holistic (complete state) information in a single event.
+
+#### Error Event: `vaFileInputError`
+
+This event is emitted when a file size or file type validation fails in any `va-file-input` child component.
+
+```typescript
+{
+  index: number,  // The index of the file that triggered the error
+  message: string // The error message
+}
+```
+
+### Event Flow & Triggers
+
+The `va-file-input-multiple` component listens to three types of `va-file-input` component events:
+
+#### File Change: `vaChange`
+
+This event is emitted when a file is selected, changed, or removed in any `va-file-input` child component.
+
+- **Trigger**: User selects, changes, or removes a file
+- **Action**:
+  - `FILE_UPDATED`: If `newFile` exists and `fileObject.file` exists
+  - `FILE_ADDED`: If `newFile` exists and `fileObject.file` is null
+  - `FILE_REMOVED`: If no `newFile` (deletion)
+
+#### Password Change: `vaPasswordChange`
+
+This event is emitted when a password is entered or changed in any `va-file-input` child component.
```

Comment:
@jamigibbs  I'm not sure if it's worth clarifying here that the event fires on every keystroke of the password being typed -- the implication for users is they probably need to debounce the event